test/integration/api/personal/access_tokens_test.rb

```diff
@@ -0,0 +1,197 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Admin::Api::Personal::AccessTokensTest < ActionDispatch::IntegrationTest
+  def setup
+    @provider = FactoryBot.create(:simple_provider)
+    host! @provider.self_domain
+    @admin = FactoryBot.create(:simple_admin, account: @provider)
+    @admin_access_token = FactoryBot.create(:access_token, owner: @admin, scopes: %w[account_management])
+  end
+
+  class ActionsOnAnAccessToken < Admin::Api::Personal::AccessTokensTest
+    def setup
+      super
+      @access_token_object = FactoryBot.create(:access_token, owner: @admin, scopes: %w[finance])
+    end
+
+    test 'access by a non-existent ID or value responds with not_found' do
+      return unless respond_to?(:perform_request)
+
+      perform_request(different_params: {id: 'wrong'})
+      assert_response :not_found
+    end
+
+    test 'perform request for a token of another user responds with not_found' do
+      return unless respond_to?(:perform_request)
+      access_token_object_admin = FactoryBot.create(:access_token, owner: @admin, scopes: %w[finance])
+      access_token_another_admin = FactoryBot.create(:access_token, scopes: %w[account_management],
+                                                     owner: FactoryBot.create(:admin, account: @provider, admin_sections: [:partners]))
+
+      access_token_object_admin.slice(:id, :value).each_value do |id_or_value|
+        perform_request(authentication: {access_token: access_token_another_admin.value}, different_params: {id: id_or_value})
+        assert_response :not_found
+      end
+    end
+
+    test 'perform request by ID works well' do
+      return if !respond_to?(:perform_request) || !respond_to?(:assert_it_worked)
```

Comment:
Why is that?

Reply:
OK I understand now. It looks like shared examples of rspec
You can also create the method in the sub-classes


```ruby
def perform_request
  skip 'some reason'
end
```